kstatus/status/generic.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
+
+import (
+	"fmt"
+
+	corev1 "sigs.k8s.io/kustomize/pseudo/k8s/api/core/v1"
+	"sigs.k8s.io/kustomize/pseudo/k8s/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func checkGenericProperties(u *unstructured.Unstructured) (*Result, error) {
+	obj := u.UnstructuredContent()
+
+	// Check if the resource is scheduled for deletion
+	deletionTimestamp, found, err := unstructured.NestedString(obj, "metadata", "deletionTimestamp")
+	if err != nil {
+		return nil, fmt.Errorf("error looking up metadata.deletionTimestamp from resource: %v", err)
```

Comment:
use `errors.New()` to get a stack trace attached

Reply:
errors.Wrap(err, "looking up metadata.deletionTimestamp from resource")

import  "github.com/pkg/errors"